junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -286,7 +291,11 @@ static CloseablePath createTempDir(TempDirFactory factory, CleanupMode cleanupMo
 
 	private static boolean selfOrChildFailed(ExtensionContext context) {
 		return context.getExecutionException().isPresent() //
-				|| context.getStore(NAMESPACE).getOrDefault(CHILD_FAILED, Boolean.class, false);
+				|| getContextSpecificStore(context).getOrDefault(CHILD_FAILED, Boolean.class, false);
+	}
+
+	private static ExtensionContext.Store getContextSpecificStore(ExtensionContext context) {
+		return context.getStore(NAMESPACE.append(context));
```

Comment:
:speech_balloon: This is the important bit: by including the context in the namespace, we prevent it from looking up the key on the parent

Reply:
Sounds reasonable.